specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/stable/2024-06-30/cloudhsm.json

```diff
@@ -0,0 +1,1441 @@
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "readOnly": true
+        }
+      },
+      "required": [
+        "privateLinkServiceConnectionState"
+      ],
+      "description": "Properties of the private endpoint connection."
+    },
+    "PrivateEndpointConnectionProvisioningState": {
+      "type": "string",
+      "readOnly": true,
+      "description": "The current provisioning state.",
+      "enum": [
+        "Succeeded",
+        "Creating",
+        "Deleting",
+        "Failed",
+        "Updating",
+        "InternalError",
+        "Canceled"
+      ],
+      "x-ms-enum": {
+        "name": "PrivateEndpointConnectionProvisioningState",
+        "modelAsString": true
+      }
+    },
+    "PrivateEndpointConnectionListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "Array of private endpoint connections.",
+          "items": {
+            "$ref": "#/definitions/PrivateEndpointConnection"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of private endpoint connections."
+        }
+      },
+      "description": "List of private endpoint connections associated with the specified resource."
+    },
+    "BackupRequestProperties": {
+      "type": "object",
+      "description": "Backup properties",
+      "allOf": [
+        {
+          "$ref": "#/definitions/BackupRestoreRequestBaseProperties"
+        }
+      ]
+    },
+    "RestoreRequestProperties": {
+      "type": "object",
+      "description": "Cloud Hsm Cluster restore information",
+      "properties": {
+        "backupId": {
+          "type": "string",
+          "description": "The ID of the backup."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/BackupRestoreRequestBaseProperties"
+        }
+      ]
+    },
+    "BackupRestoreRequestBaseProperties": {
+      "type": "object",
+      "description": "Backup and Restore Common properties",
+      "properties": {
+        "azureStorageBlobContainerUri": {
+          "type": "string",
+          "description": "The Azure blob storage container Uri which contains the backup"
+        },
+        "token": {
+          "type": "string",
+          "description": "The SAS token pointing to an Azure blob storage container. This property is reserved for Azure Backup Service."
+        }
+      },
+      "required": [
+        "azureStorageBlobContainerUri"
+      ]
+    },
+    "BackupResult": {
+      "type": "object",
+      "properties": {
+        "azureStorageBlobContainerUri": {
+          "type": "string",
+          "description": "The Azure blob storage container Uri which contains the backup"
+        },
+        "backupId": {
+          "type": "string",
+          "description": "The ID of the backup."
```

Comment:
>The ID of the backup. [](http://example.com/codeflow?start=26&length=21)

Please add to description indicating type of ID: e.g. resource ID, GUID, name, something else. Use format: uuid, or arm-id, as the case may be.

Reply:
The backup Id will be a combination of a static string, GUID and a date: cloudhsm-e1d2f18e-c919-4b44-b352-363f3a41f16c-2024041615425595

Reply:
Can I suggest updating the description to something like 
"An autogenerated unique string ID for labeling the backup. It contains both a UUID and a date timestamp."

Reply:
updated thanks !